src/containers/ViewerPanel/index.tsx

```diff
@@ -431,6 +462,62 @@ class ViewerPanel extends React.Component<ViewerPanelProps, ViewerPanelState> {
         }
     };
 
+    private updateDisplayBreakpoints = () => {
+        if (location.pathname === EMBED_PATHNAME) {
+            this.setState({
+                embedDisplayBreakpoints: {
+                    belowControlsHeight:
+                        window.innerHeight <=
+                        EMBEDDED_STANDARD_CONTROLS_MINIMUM_HEIGHT,
+                    belowControlsWidth:
+                        window.innerWidth <=
+                        EMBEDDED_STANDARD_CONTROLS_MINIMUM_WIDTH,
+                    belowScaleBarWidth:
+                        window.innerWidth <= EMBEDDED_SCALE_BAR_MINIMUM_WIDTH,
+                },
+            });
+        }
+    };
+
+    private handleFullscreenChange = () => {
+        const { setEmbedFullscreen } = this.props;
+        setEmbedFullscreen(!!document.fullscreenElement);
+        const current = this.centerContent.current;
+        if (current) {
+            setTimeout(() => {
+                this.resize(current);
+                this.updateDisplayBreakpoints();
+            }, 100);
+        }
+    };
+
+    private toggleFullscreen = () => {
+        const { embedFullscreen } = this.props;
+        if (embedFullscreen) {
+            document.exitFullscreen();
+        } else {
+            document.documentElement.requestFullscreen().catch((err) => {
+                console.error(
+                    `Error attempting to enable fullscreen: ${err.message}`
+                );
+            });
+        }
+    };
+
+    private get embedDisplaySettings() {
+        const { belowControlsHeight, belowControlsWidth, belowScaleBarWidth } =
+            this.state.embedDisplayBreakpoints;
+        return {
+            minimalPlaybackControls: belowControlsWidth,
+            minimalCameraControls: belowControlsHeight,
+            onlyRenderBottomControls: belowControlsWidth && belowControlsHeight,
+            showScaleBar:
+                !belowScaleBarWidth &&
+                (!belowControlsWidth || !belowControlsHeight),
+            showCameraControls: !belowControlsWidth || !belowControlsHeight,
```

Comment:
I'm trying to follow this, but it's really confusing: partially because you have some things that are in the minimize direction, and some that are in opposite. So you have both ! and ors and ands. I'm going to take a stab at simplifying this

Reply:
https://github.com/simularium/simularium-website/pull/568